### PR TITLE
Add raw mouse input option (new approach)

### PR DESCRIPTION
**Pull request by dreamer**
From commit message:

> SDL defaults to using mouse input processed by the OS. This makes
> mouse cursor have the same acceleration as rest of the system.
> 
> However, some games (e.g. FPS titles, like DOOM or Dark Forces) will
> work better with raw input (especially modern gaming mouses capable
> of very high DPI settings).
> 
> When SDL is instructed to use raw mouse input, it affects fullscreen
> and windows that captured the mouse cursor.  When window is configured
> with capture_mouse=seamless, then raw input is ignored (which is good).
> 
> Unfortunately, receiving raw input events depends on the OS - there's
> history of Windows 10 updates breaking this feature, and support on
> Linux seems generally broken as of 2020.
> 
> Keep new option disabled by default, as we don't want to affect games,
> where OS-native mouse behaviour would be more preferable (strategy
> games, adventure games, etc).
> 
> This feature requires SDL 2.0.2
> 
> Fixes: #244

I started with the code written @NicknineTheEagle in #246 but made some maintenance-related modifications and documentation changes. I tested it on Windows 10 and on Linux.

- On Linux the feature seems to not work (possible libinput bug, hard to tell).
- On Windows 10 it works well, but to see experience any difference I needed to make my cursor speed in Windows unusably slow (I don't have a gaming mouse, which is much more likely to show the difference for this option)

@NicknineTheEagle, @kcgen please look at this modified version - I hope new option description is a bit friendlier towards non-technical users.

### CODE REVIEW

File: src/gui/sdlmain.cpp [hunk 1]
```diff
@@ -2876,6 +2881,11 @@ void Config_Add_SDL() {
 	Pint = Pmulti->GetSection()->Add_int("ysens",Property::Changeable::Always,100);
 	Pint->SetMinMax(-1000,1000);
 
+	pbool = sdl_sec->Add_bool("raw_mouse_input", on_start, false);
+	pbool->Set_help("Enable this setting to bypass mouse sensitivity and\n"
```
Comment (dreamer) [inline src/gui/sdlmain.cpp:2885]: @kcgen I am not sure about using the word "bypass" in here - perhaps "disable" would work better? How does this sentence read to a native English speaker?

File: src/gui/sdlmain.cpp [hunk 2]
```diff
@@ -2876,6 +2881,11 @@ void Config_Add_SDL() {
...
+	                "acceleration profile used by your operating system. This\n"
+	                "works in fullscreen and in window with captured cursor.");
```
Comment (kcgen) [inline src/gui/sdlmain.cpp:2887]: *"... works in fullscreen or when the cursor is captured."* I think that reads a bit better?

File: src/gui/sdlmain.cpp (hunk 1 above)
Comment (kcgen) [reply src/gui/sdlmain.cpp:2885]: Bypass sounds good to me. It's the OS that's being bypassed, so I can only suggestion something like: *"Enable this setting to bypass your operating system's mouse sensitivity and acceleration settings."*

**Review by kcgen — APPROVED** [review]
Only wording suggestions; looks good and a nice refinement regarding version handling.
Although thumbs up to the co-authoring (expected, but just wanted to 'ack it!). 
Nice work @NicknineTheEagle and @dreamer.

File: src/gui/sdlmain.cpp (hunk 1 above)
Comment (NicknineTheEagle) [reply src/gui/sdlmain.cpp:2885]: I think this description is fine.

Comment (kcgen) [reply src/gui/sdlmain.cpp:2885]: Not sure where my prior comment went! 
Bypass is good; but suggest a small order change to the active voice: *"Enable this setting to bypass your operating system's mouse acceleration and sensitivity settings."*

**Comment by dreamer** [comment]
I changed the description (I think this is the most non-technical one we came up so far).

Did some Googling to see if I'll find any reason for this setting not working, and there's some good news and some worrying news:

Good news: after replacing `SDL_SetHint` with `SDL_SetHintWithPriority(..., SDL_HINT_OVERRIDE);`, the setting now seems to work (I am not 100% sure). When setting sensitivity to the lowest setting in Gnome/X11, I could hardly see the difference in fullscreen (so the same testing procedure I used on Windows was inconclusive). However, when I set the sensitivity to the max, I *think* I can feel the difference (cursor has faster acceleration with `raw_mouse_input = false` and flatter response curve with `raw_mouse_input = true`). But this still might be a placebo effect, I have no idea how to actually measure it.

Worrying news: enabling raw mouse input seems to be often broken on *all* operating systems. Windows 10 updates sometimes break it, macOS does not like the setting - it seems like it conflicts with accessibility settings and it's hard to tell if it works on modern Linux… I think we should expect 2-3 follow-up bugs.

That being said - when it works, I expect it will be greatly appreciated.

**Review by kcgen — APPROVED** [review]
Nothing more to say on my side; let's merge it!

**Review by kcgen — APPROVED** [review]
I was unable to tell the difference between raw and without on Linux (even using standalone code, back when this originally was submitted), but if you can feel even a slight change then that's better than nothing!

**Comment by dreamer** [comment]
@kcgen Thanks for the review!
@NicknineTheEagle Thanks for pushing for inclusion of this feature :)

**Pull request merged**